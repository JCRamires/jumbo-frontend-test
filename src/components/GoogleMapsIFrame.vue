<script setup lang="ts">
interface Props {
  latitude: string;
  longitude: string;
}

const { latitude, longitude } = defineProps<Props>();
</script>

<template>
  <div class="iframe-container">
    <iframe
      width="800"
      height="600"
      frameborder="0"
      scrolling="no"
      marginheight="0"
      marginwidth="0"
      :src="`https://maps.google.com/maps?q=${latitude},${longitude}&z=14&amp;output=embed`"
    >
    </iframe>
  </div>
</template>

<style scoped>
.iframe-container {
  position: relative;
  height: 50vh;
  overflow: hidden;
}

.iframe-container iframe {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}
</style>
