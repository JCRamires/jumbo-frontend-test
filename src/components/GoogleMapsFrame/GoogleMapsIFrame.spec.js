import { describe, it, expect } from "vitest";
import { mount } from "@vue/test-utils";
import GoogleMapsIFrame from "./GoogleMapsIFrame.vue";

describe("GoogleMapsIFrame", () => {
  it("mounted component should match snapshot", async () => {
    const wrapper = mount(GoogleMapsIFrame, {
      props: {
        latitude: "123",
        longitude: "321",
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
