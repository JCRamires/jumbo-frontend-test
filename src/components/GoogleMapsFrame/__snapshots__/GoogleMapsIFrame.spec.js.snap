// Vitest Snapshot v1

exports[`GoogleMapsIFrame > mounted component should match snapshot 1`] = `
<div
  class="iframe-container"
  data-v-a826f365=""
>
  <iframe
    data-v-a826f365=""
    frameborder="0"
    height="600"
    marginheight="0"
    marginwidth="0"
    scrolling="no"
    src="https://maps.google.com/maps?q=123,321&z=14&output=embed"
    width="800"
  >
    
    
  </iframe>
</div>
`;
